framework/audio/internal/midiplayer.h

```diff
@@ -0,0 +1,128 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#ifndef MU_AUDIO_MIDIPLAYER_H
+#define MU_AUDIO_MIDIPLAYER_H
+
+#include <memory>
+#include <vector>
+#include <map>
+#include <cstdint>
+#include <functional>
+#include <mutex>
+
+#include "imidiplayer.h"
+#include "modularity/ioc.h"
+#include "async/asyncable.h"
+#include "midi/isynthesizersregister.h"
+#include "midi/imidiportdatasender.h"
+
+namespace mu {
+namespace audio {
+class MIDIPlayer : public IMIDIPlayer, public async::Asyncable
+{
+    INJECT(midi, midi::ISynthesizersRegister, synthesizersRegister)
+    INJECT(midi, midi::IMidiPortDataSender, midiPortDataSender)
+
+public:
+    MIDIPlayer();
+    ~MIDIPlayer() override;
+
+    void loadMIDI(const std::shared_ptr<midi::MidiStream>& stream) override;
+    void forwardTime(unsigned long miliseconds) override;
+
+    void run() override;

```

Comment:
better to name start because you have stop method

Reply:
no, better play

Reply:
@Eism renamed respectively `IPlayer::play()` Thanks